_docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Service/WordpressApi.php

```diff
@@ -147,40 +209,67 @@ public function getAutocompleteContentOptions($search, $max_results) {
       'search' => $search,
     ];
     $results = [];
+    $cid = 'wordpress_api:search:' . $search;
 
-    try {
-      // Tries to retrieve WP posts from $feed_url using the $query params.
-      $request = $this->client->request('GET', $feed_url, ['query' => $query]);
-      $response = $request->getBody()->getContents();
-      $api_results = json_decode($response);
-
-      foreach ($api_results as $api_result) {
-        $label = Html::escape($api_result->post_title) . ' (' . $api_result->ID . ')';
-        $results[] = [
-          'value' => $label,
-          'label' => $label,
-        ];
-
-        if (count($results) >= $max_results) {
-          // If we already have $max_results-many WP posts in $results, then we
-          // return $results.
-          return $results;
-        }
+    // Serve the cached Wordpress API data if available.
+    if ($cache = $this->cacheBin->get($cid)) {
+      $data = $cache->data;
+    }
+    // If this is uncached or the cache is invalid, fetch fresh data.
+    else {
+      try {
+        // Tries to retrieve WP posts from $feed_url using the $query params.
+        $request = $this->client->request('GET', $feed_url, ['query' => $query]);
+        $response = $request->getBody()->getContents();
+        $api_results = json_decode($response);
+        $data = $api_results;
```

Comment:
Why are these lines slightly different than lines 168-177?  Can this be functional?

Reply:
That is a really good point. I think that this could be extracted into a function to make this solution more DRY.

Reply:
I assume that depends on the time we expect to utilize WP api calls?  I wouldn't say this is necessary for this PR, and my comment can be ignored. :)